pkg/tasks/prometheus_user_workload.go

```diff
@@ -44,6 +44,7 @@ func (t *PrometheusUserWorkloadTask) Run(ctx context.Context) error {
 		return t.create(ctx)
 	}
 
+	klog.V(3).Infof("UWM prometheus is disabled (because UWM is disabled), existing related resources are to be destroyed.")
```

Comment:
how to check the logs? checked all pod logs under `openshift-monitoring`, did not find it

Reply:
I should have been more explicit than in https://github.com/openshift/cluster-monitoring-operator/pull/2546#issue-2756512895

By default, CMO runs with log level `--v=2` https://github.com/openshift/cluster-monitoring-operator/blob/3d6c61fa3c35d83b25d8d26e57c35dcd359d1da4/manifests/0000_50_cluster-monitoring-operator_05-deployment.yaml#L59
That should be set to at least `3` to see the logs. (requires setting CMO to unmanaged, I expect anyone who wants to learn more about CMO's decisions to increase the verbosity level)

Logging at `V(2)` would be too verbose for normal operations

As an alternative, I considered making this more visible via the monitoring `co` status:
```
  - lastTransitionTime: "2024-12-26T13:04:36Z"
    message: "YYY component disabled etc."
    status: "False"
    type: Degraded
```
However, I think it would make more sense to include this in the future monitoring CR itself. I'll ensure we discuss this and take it into account.

Reply:
found the info, steps
```
$ oc -n openshift-cluster-version scale deploy cluster-version-operator --replicas=0
$ oc -n openshift-monitoring scale deploy cluster-monitoring-operator --replicas=0
```
then edit deploy cluster-monitoring-operator, change to -v=3 and scale deploy cluster-monitoring-operator=1
```
$ oc -n openshift-monitoring get deploy cluster-monitoring-operator -oyaml
      - args:
       ...
        - -v=3

$ oc -n openshift-monitoring scale deploy cluster-monitoring-operator --replicas=1
 ```
```
oc -n openshift-monitoring logs deploy/cluster-monitoring-operator
I1231 03:21:21.533009       1 thanos_ruler_user_workload.go:46] UWM thanos ruler is disabled (because UWM is disabled), existing related resources are to be destroyed.
I1231 03:21:21.533009       1 tasks.go:70] running task 2 of 15: UpdatingPrometheus
I1231 03:21:21.533018       1 tasks.go:70] running task 1 of 15: UpdatingClusterMonitoringOperatorDeps
I1231 03:21:21.533027       1 tasks.go:70] running task 5 of 15: UpdatingKubeStateMetrics
I1231 03:21:21.533035       1 tasks.go:70] running task 11 of 15: UpdatingConsolePluginComponents
I1231 03:21:21.533039       1 tasks.go:70] running task 3 of 15: UpdatingAlertmanager
I1231 03:21:21.533083       1 tasks.go:70] running task 8 of 15: UpdatingTelemeterClient
I1231 03:21:21.533135       1 tasks.go:70] running task 9 of 15: UpdatingThanosQuerier
I1231 03:21:21.533118       1 telemeter.go:56] Telemeter client is disabled (because telemetry explicitly is disabled), existing related resources are to be destroyed.
I1231 03:21:21.533222       1 tasks.go:70] running task 4 of 15: UpdatingNodeExporter
I1231 03:21:21.533247       1 tasks.go:70] running task 13 of 15: UpdatingUserWorkloadPrometheus
I1231 03:21:21.533254       1 prometheus_user_workload.go:47] UWM prometheus is disabled (because UWM is disabled), existing related resources are to be destroyed.
I1231 03:21:21.533264       1 tasks.go:70] running task 12 of 15: UpdatingUserWorkloadPrometheusOperator
I1231 03:21:21.533269       1 prometheusoperator_user_workload.go:46] UWM prometheus operator is disabled (because UWM is disabled), existing related resources are to be destroyed.
I1231 03:21:21.533279       1 tasks.go:70] running task 14 of 15: UpdatingUserWorkloadAlertmanager
I1231 03:21:21.533284       1 alertmanager_user_workload.go:51] UWM alertmanager is disabled, existing related resources are to be destroyed.
```